core/chains/evm/bulletprooftxmanager/eth_broadcaster.go

```diff
@@ -345,20 +359,36 @@ func (eb *EthBroadcaster) handleInProgressEthTx(etx EthTx, attempt EthTxAttempt,
 	}
 	parentCtx := context.TODO()
 
-	if etx.Simulate {
-		simulationCtx, cancel := context.WithTimeout(parentCtx, SimulationTimeout)
-		defer cancel()
-		if b, err := simulateTransaction(simulationCtx, eb.ethClient, attempt, etx); err != nil {
-			if jErr := evmclient.ExtractRPCError(err); jErr != nil {
-				eb.logger.CriticalW("Transaction reverted during simulation", "ethTxAttemptID", attempt.ID, "txHash", attempt.Hash, "err", err, "rpcErr", jErr.String(), "returnValue", b.String())
-				etx.Error = null.StringFrom(fmt.Sprintf("transaction reverted during simulation: %s", jErr.String()))
-				return eb.saveFatallyErroredTransaction(&etx)
-			}
-			eb.logger.Warnw("Transaction simulation failed, will attempt to send anyway", "ethTxAttemptID", attempt.ID, "txHash", attempt.Hash, "err", err, "returnValue", b.String())
-		} else {
-			eb.logger.Debugw("Transaction simulation succeeded", "ethTxAttemptID", attempt.ID, "txHash", attempt.Hash, "returnValue", b.String())
-		}
+	checkerSpec, err := etx.GetChecker()
+	if err != nil {
+		return errors.Wrap(err, "parsing transmit checker")
+	}
+
+	checker, err := eb.checkerFactory.BuildChecker(checkerSpec)
+	if err != nil {
+		return errors.Wrap(err, "building transmit checker")
+	}
+
+	// If the transmit check does not complete within 2 seconds, the transaction will be sent
+	// anyway.
+	checkCtx, cancel := context.WithTimeout(parentCtx, 2*time.Second)
```

Comment:
Did this used to be a named constant? It seems to have turned into a magic number in the code now.

It would be even better to have it configurable as an ENV var (just in case).

Reply:
Yup moved back to a constant. An env var SGTM but would rather do it in a separate PR since this is already pretty big and touching a lot of code.